tools/package-docs-content/index.bzl

```diff
@@ -4,22 +4,23 @@
 """
 
 def _package_docs_content(ctx):
-    # Directory that will contain all grouped input files. This directory will be created
-    # relatively to the current target package. (e.g. "bin/src/components-examples/docs-content")
-    output_dir = ctx.attr.name
-
     # Arguments that will be passed to the packager executable.
     args = ctx.actions.args()
 
-    # List of outputs that should be generated by the packager action. Bazel will automatically
-    # throw an error if any output has not been generated properly.
-    expected_outputs = []
+    # Directory that will contain all grouped input files. This directory will be
+    # created relatively to the current target package. For example:
+    # "bin/src/components-examples/docs-content/docs-content". The reason we need to
+    # repeat `docs-content` is for
```

Comment:
ahhhh! What's the reason? Please help :)

Reply:
Yeah, we should complete that comment (that was actually my bad; I sent an incomplete patch). It's basically the same issue as mentioned in my first review comment. We should link those together by referring to a dedicated JIRA I guess.

Reasoning for putting the docs-content into another `docs-content` folder is that the `ng_package` rule does not properly handle tree artifacts in `data`. The trick is that we create a tree artifact that can be put into `nested_packages`. Nested packages do not preserve the tree artifact name (i.e. the directory name), so all contents of the docs-content would be put directly into the `@angular/components-examples` package. We don't want that, but preserve the `docs-content/` folder. So we have another folder like `docs-content` in the tree artifact that is preserved as content of the tree artifact. Not really nice. I know.